### PR TITLE
KAA-1147: Extend C++ page

### PR DESCRIPTION
**Pull request by RostakaGmfun**
/cc @akokhanovskyi @forGGe

### CODE REVIEW

**Comment by forGGe** [comment]
+

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/index.md [hunk 1]
```diff
@@ -9,23 +9,32 @@ sort_idx: 20
...
+
+In case you need to run Kaa endpoints on bare metal hardware, consider using [Kaa C SDK]({{root_url}}/Programming-guide/Using-Kaa-endpoint-SDKs/C).
+
+For in-depth architecture overview, refer to [Architecture overiew]({{root_url}}/Customization-guide/Endpoint-SDKs/C++-SDK/Architecture-overview/) page.
 
-This page describes build configuration for Kaa C++ SDK.
-For detailed installation instructions for different platforms, check the corresponding subpages.
+## Environment setup
+
+The recommended way to setup required dependencies is by using [Nix package mananger]({{root_url}}/Customization-guide/Endpoint-SDKs/C-SDK/Environment-setup/Nix-guide).
```
Comment (rasendubi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/index.md]: Nix is not recommended for the end user. The user should install all dependencies using the native method for his platform. Nix is nice for developing the SDK itself, but not for building end applications.

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/index.md [hunk 2]
```diff
@@ -9,134 +9,142 @@ sort_idx: 20
...
 
-This page describes build configuration for Kaa C++ SDK.
-For detailed installation instructions for different platforms, check the corresponding subpages.
+In case you need to run Kaa endpoints on bare metal hardware, consider using [the C SDK]({{root_url}}/Programming-guide/Using-Kaa-endpoint-SDKs/C).
+
+For in-depth architecture overview, refer to [C++ SDK architecture overview page]({{root_url}}/Customization-guide/Endpoint-SDKs/C++-SDK/Architecture-overview/) page.
+
+## Environment setup
+
+The recommended way to setup required dependencies is by using [Nix package mananger]({{root_url}}/Customization-guide/Endpoint-SDKs/C-SDK/Environment-setup/Nix-guide).
```
Comment (rasendubi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/index.md]: Nix is not recommended for the end user. The user should install all dependencies using the native method for his platform. Nix is nice for developing the SDK itself, but not for building end applications.

**Comment by RostakaGmfun** [comment]
@rasendubi re-review please

**Comment by rasendubi** [comment]
LGTM

**Comment by forGGe** [comment]
+

**Pull request merged**